R/PipeOpFilterRows.R

```diff
@@ -0,0 +1,171 @@
+#' @title PipeOpFilterRows
+#'
+#' @usage NULL
+#' @name mlr_pipeops_filterrows
+#' @format [`R6Class`] object inheriting from [`PipeOpTaskPreproc`].
+#'
+#' @description
+#' Filter rows of the data of a task. Also directly allows for the removal of rows holding missing
+#' values.
+#'
+#' @section Construction:
+#' ```
+#' PipeOpFilterRows$new(id = "filterrows", param_vals = list())
+#' ```
+#'
+#' * `id` :: `character(1)`\cr
+#'   Identifier of resulting object, default `"filterrows"`.
+#' * `param_vals` :: named `list`\cr
+#'   List of hyperparameter settings, overwriting the hyperparameter settings that would otherwise
+#'   be set during construction. Default `list()`.
+#'
+#' @section Input and Output Channels:
+#' Input and output channels are inherited from [`PipeOpTaskPreproc`].
+#'
+#' The output during training is the input [`Task`][mlr3::Task] with rows kept according to the
+#' filtering (see Parameters) and (possible) rows with missing values removed.
+#' 
+#' The output during prediction is the unchanged input [`Task`][mlr3::Task] if the parameter
+#' `skip_during_predict` is `TRUE`. Otherwise it is analogously handled as the output during
+#' training.
+#'
+#' @section State:
+#' The `$state` is a named `list` with the `$state` elements inherited from [`PipeOpTaskPreproc`],
+#' as well as the following elements:
+#' * `na_ids` :: `integer`\cr
+#'   The row identifiers that had missing values during training and therefore were removed. See the
+#'   parameter `na_column`.
+#' * `row_ids` :: `integer`\cr
+#'   The row identifiers that were kept during training according to the parameters `filter`,
+#'   `na_column` and `invert`.
+#'
+#' @section Parameters:
+#' The parameters are the parameters inherited from [`PipeOpTaskPreproc`], as well as:
+#' * `filter` :: `NULL` | `character(1)` | `expression` | `integer`\cr
+#'   How the rows of the data of the input [`Task`][mlr3::Task] should be filtered. This can be a
+#'   character vector of length 1 indicating a feature column of logicals in the data of the input
+#'   [`Task`][mlr3::Task] which forms the basis of the filtering, i.e., all rows that are `TRUE`
+#'   with respect to this column are kept in the data of the output [`Task`][mlr3::Task]. Moreover,
+#'   this can be an expression that will result in a logical vector of length `$nrow` of the data of
+#'   the input [`Task`][mlr3::Task] when evaluated withing the environment of the `$data()` of the
+#'   input [`Task`][mlr3::Task]. Finally, this can also be an integerish vector that directly
+#'   specifies the row identifiers of the rows of the data of the input [`Task`][mlr3::Task] that
+#'   should be kept. Default is `NULL`, i.e., no filtering is done.
+#' * `na_column` :: `NULL` | `character`\cr
+#'   A character vector that specifies the columns of the data of the input [`Task`][mlr3::Task]
+#'   that should be checked for missing values. If set to `all`, all columns of the data are used. A
```

Comment:
can we somehow use `all` that is unlikely to be a colname? `_all_` or something?

Reply:
What have we done in other `PipeOp`s here?

Reply:
PipeOpMissingIndicators.R uses `"all"` but I do like the suggestion of `"_all_"`.